Feature/new-drs-object

ElizabethJanes opened the pull request:
Implemented NewDrs page which has custom functionality to allow user to populate DRS Object fields. By clicking the submit button on this page, a POST request is made to add a new DRS Object to the database.

ElizabethJanes committed:
Implement CreateDrsForm, update router and buttons

ElizabethJanes committed:
Update CreateDrsForm

ElizabethJanes committed:
Implement resetActiveDrsObject

ElizabethJanes committed:
Initialize newDrsObject

ElizabethJanes committed:
Implement update property functions

ElizabethJanes committed:
Update string properties

ElizabethJanes committed:
Implement NewDrs page and implement radio to select bundle or blob when creating new DRS Object

ElizabethJanes committed:
Implement add button for list properties

ElizabethJanes committed:
Refactor update DRS property functions

ElizabethJanes committed:
Refactor and implement functionality to add new list item when creating new DRS object

ElizabethJanes committed:
Implement functionality for adding and modifying list properties when creating new DRS object

ElizabethJanes committed:
Update button styling and implement dropdown selector on DrsObjectForm

ElizabethJanes committed:
Implement remove button on NewDrs form

ElizabethJanes committed:
Refactor DrsObjectForm, update position of "remove alias" button, and implement date-time handling

ElizabethJanes committed:
Implement "Generate Random UUID" button and update date-time pickers

ElizabethJanes committed:
Update select for Checksum Type so that options are disabled if they have already been selected

ElizabethJanes committed:
Refactor using boolean type identifiers and update DrsObjectForm formatting

ElizabethJanes committed:
Populate name field after entering ID of parent and child objects

ElizabethJanes committed:
Refactor, update 'VerifyId' button, and begin implementing submit button functionality

ElizabethJanes committed:
Implement POST request cleanup function, update date-time selectors, implement error handling, and update 'Verify Id' button

ElizabethJanes committed:
Implement navigation buttons, update DrsObjectForm for edge cases, prevent refresh on each route change

ElizabethJanes committed:
Refactor, implement componentDidUpdate method,

ElizabethJanes committed:
Update date-time handling

ElizabethJanes committed:
Refactor API requests, update Index page upon navigation, handle invalid properties when adding new DRS

ElizabethJanes committed:
Implement redirect after form submission and update commenting

ElizabethJanes committed:
Refactor, resolve issues related to adding new DRS Object, and update code to handle is_bundle property which is now submitted to and received from DRS Starter Kit API requests

ElizabethJanes committed:
Update some DrsObjectForm logic and use a hook to update active DRS Object when navigating to NewDrs page

jb-adams commented on src/components/drs/Drs.js:
```diff
@@ -3,72 +3,291 @@ import axios from 'axios';
 
-const cancelToken = axios.CancelToken;
-const drsCancelToken = cancelToken.source();
+let newDate = new Date();
```
Do you need to unpack all attributes of a Date for the constructor below? Why not just do:

```
let newDate = new Date();
newDate.setSeconds(0, 0);
...
created_time: format(newDate, "yyyy-MM-dd'T'HH:mm:ss'Z'"),
```

jb-adams commented on src/components/drs/Drs.js:
```diff
@@ -3,72 +3,291 @@ import axios from 'axios';
+    this.updateSubmitNewDrsRedirect = this.updateSubmitNewDrsRedirect.bind(this);
     this.state = {
-      activeDrsObject: null,
+      activeDrsObject: {
```
this code of creating a "blank" DrsObject is written both here and in`drs/pages/NewDrs.js` almost exactly the same. Do both components need to run it? If so, define a single function and pass it to wherever needs to run it. It is essentially a `resetActiveDrsObject` function

jb-adams commented on src/components/drs/pages/DrsShow.js:
```diff
@@ -1,84 +1,99 @@
 import '@fontsource/roboto';
```
do you really need `DrsShow` and `NewDrs` anymore? The returned JSX from each is almost exactly the same (only the title appears different, which can be parameterized and added to `DrsObjectForm`).

The custom logic of updating the `activeDrsObject` based on API calls can reside in the central `Drs` class, since it holds the `activeDrsObject` in the first place

jb-adams committed:
moving drs form components into separate files, update functions defined in root drs class, individual functions passed to sub-components, etc

jb-adams committed:
merge and group props passed to form by uptaking component, refactored DrsObject children and parents components

jb-adams committed:
modularize and update file and aws access point form

jb-adams committed:
restored POST request and pre-validation

jb-adams committed:
apply conditional visualization to all props, successful POST request triggers reload of DrsObjects from REST API, and redirects to listing page

jb-adams committed:
remove 'VerifyId' button from the SHOW page for DRS children and parents

jb-adams committed:
address items 1,2,4,5,6,9
- "View relative" button added
- scroll to top upon id change
- empty id when submitted for verification returns invalid
- relative 'name' removed upon change
- an object's parent must be bundle-type

jb-adams committed:
fix handling of dates between local TZ and UTC

jb-adams committed:
remove later checksums if previous checksum type is changed to overwrite it

jb-adams committed:
correct handling of timestamp initial state for new DrsObjects, fix parent bundles being marked as invalid, page refresh and formatting

ElizabethJanes committed:
Merge pull request #1 from jb-adams/feature/create-drs-object-jeremy

refactor of the form to create DrsObject and other components

ElizabethJanes committed:
Update packages to align with updates from merging pull request
